### ISSUE TITLE
Add Black and format all python files

### ISSUE THREAD

**Issue by gmotzespina** [neuropoly/spinalcordtoolbox#2439]
**Feature justification: Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Add any other context (references, screenshots...) about the feature request here.

**Comment by jcohenadad** [comment]
Before reformatting files, we should first agree on conventions. The major convention concerns single vs. double quotes. I kind of like the idea of having single quote for variables, and double quotes for verbose text, which seems to be a popular way, see for instance [this post](https://stackoverflow.com/a/56190).

**Comment by gmotzespina** [comment]
I like the convention you mentioned as well. 

* single quote for variables 
* double quotes for verbose text

**Comment by jcohenadad** [comment]
Others, any comments?  @zougloub @charleygros @perone 

@gmotzespina Once we agree on the convention, please don't forget to update [the wiki](https://github.com/neuropoly/spinalcordtoolbox/wiki/Single-or-double-quotes).

**Comment by charleygros** [comment]
Although I knew about `flake8` etc., I did not know about `black`. Did a bit of research about it and do not really have other comments than... sounds really great and useful! Help us to focus our code reviews on content.

**Comment by gmotzespina** [comment]
@zougloub do you have any other thoughts ? 

So according to PEP 8 https://www.python.org/dev/peps/pep-0008/#string-quotes

There is no recommendation on wheather to use single or double quotes. 

> In Python, single-quoted strings and double-quoted strings are the same. This PEP does not make a recommendation for this. Pick a rule and stick to it. When a string contains single or double quote characters, however, use the other one to avoid backslashes in the string. It improves readability.

> For triple-quoted strings, always use double quote characters to be consistent with the docstring convention in PEP 257.

So maybe we should:

* Use **double** quotes for **natural language messages** 
* Use **double** quotes on **triple-quoted strings**
* Use **single** quotes on **simple string variables**
* Use **double** quotes whenever a variable has some text which **requires a single quote**

**Comment by jcohenadad** [comment]
@gmotzespina make sure we can tweek black to have single quotes in some cases. see [here](http://books.agiliq.com/projects/essential-python-tools/en/latest/linters.html#black):

> Black is not configurable(except for line-length).

**Comment by gmotzespina** [comment]
@jcohenadad You're right. 

I think that leaves us with the following options, 

* use automatic black formatting (double quotes)
* use automatic black formatting and manually adjusting the code to our needs
* find another file formatter which allows us to configure it to our needs
* do not use a file formatter and just try to be careful and follow our code standards.

**Comment by jcohenadad** [comment]
Related to https://github.com/neuropoly/spinalcordtoolbox/issues/1538

**Comment by cfhammill** [comment]
I don't think we necessarily need to have automatic formatting, I think it's handy to help us get all our code up to standard, but after that we can leave compliance in the hands of the PR author and suggest using autopep8 or black if they have a big chunk of code out of compliance.
